src/liballoc/sync.rs

```diff
@@ -328,6 +328,40 @@ impl<T> Arc<T> {
         Self::from_inner(Box::leak(x).into())
     }
 
+    /// Constructs a new `Arc<T>` using a weak reference to itself. Attempting
+    /// to upgrade the weak reference before this function returns will result
+    /// in a `None` value.
+    #[inline]
+    #[unstable(feature = "arc_new_with", issue = "none")]
+    pub fn new_with(data_fn: impl FnOnce(&Weak<T>) -> T) -> Arc<T> {
+        // Start the weak pointer count at 2:
```

Comment:
The code sets the weak count to 1, though. That does not seem to match the comment.

Reply:
Fixed